microceph/ceph/operations.go

```diff
@@ -128,17 +124,29 @@ func (o *CheckNonOsdSvcEnoughOps) Run(name string) error {
 		"mgr": 0,
 		"mds": 0,
 	}
+	total_mon_services := 0
 	for _, service := range services {
-		// do not count the service on this node
+		// do not count the services on this node (these are the services remaining if this node is down)
 		if service.Location != name {
 			remains[service.Service]++
 		}
+
+		// count the total number of mon services, so we can calculate the minimum number of mon services required for quorum
+		if service.Service == "mon" {
+			total_mon_services += 1
+		}
 	}
 
+	// a majority of ceph-mon services must remain active to retain quorum
+	minMon := total_mon_services / 2 + 1
+	// only need one ceph-mds and one ceph-mgr: they operate as one active, the rest in standby
+	minMds := 1
+	minMgr := 1
+
 	// the remaining services must be sufficient to make the cluster healthy after the node enters
 	// maintanence mode.
-	if remains["mon"] < o.MinMon || remains["mds"] < o.MinMds || remains["mgr"] < o.MinMgr {
-		return fmt.Errorf("need at least %d mon, %d mds, and %d mgr services in the cluster besides those in node '%s'", o.MinMon, o.MinMds, o.MinMgr, name)
+	if remains["mon"] < minMon || remains["mds"] < minMds || remains["mgr"] < minMgr {
```

Comment:
This change can be tested in `operations_test.go` (I did not know why I did not add unit test for this function..., may be I over looked)